sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,36 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, *args):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> DiracDelta(x - 5, 4).rewrite(Piecewise)
```

Comment:
> Rewriting a delta as a piecewise is already incorrect

How should we balance the mathematical pureness with applied use cases? We are working to implement this rewrite so that we can use to express the piecewise solutions for applied engineering beam problems in the long run. There are other issues that people are arguing over how heaviside and laplace transforms should behave where the people that want to use these things for applications need it to behave not-so mathematically pure. I'm not sure I know how we should be handling these things in SymPy. Suggestions?